fashionmnist_model.py

```diff
@@ -1,54 +1,73 @@
 
+
 def create_model():
-    model = keras.Sequential([
-        layers.Dense(128, activation='relu', input_shape=(784,)),
-        layers.Dropout(0.3),
-        layers.BatchNormalization(),
-        layers.Dense(24, activation='relu'),
-        layers.Dropout(0.3),
-        layers.BatchNormalization(),
-        layers.Dense(24, activation='relu'),
-        layers.Dropout(0.3),
-        layers.BatchNormalization(),
-        layers.Dense(10, activation='softmax'),
-    ])
-    
+    model = keras.Sequential(
+        [
+            layers.Dense(128, activation="relu", input_shape=(784,)),
+            layers.Dropout(0.3),
+            layers.BatchNormalization(),
+            layers.Dense(24, activation="relu"),
+            layers.Dropout(0.3),
+            layers.BatchNormalization(),
+            layers.Dense(24, activation="relu"),
+            layers.Dropout(0.3),
+            layers.BatchNormalization(),
+            layers.Dense(10, activation="softmax"),
+        ]
+    )
+
     return model
 
-def compile_and_train(model, X_train, y_train, X_test, y_test, optimizer, batch_size=100, epochs=30, validation_split=0.2):
-    model.compile(loss="categorical_crossentropy", optimizer=optimizer, metrics=['accuracy', 'mae'])
-    history = model.fit(X_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=validation_split)
-    
+
+def compile_and_train(
+    model,
+    X_train,
+    y_train,
+    X_test,
+    y_test,
+    optimizer,
+    batch_size=100,
+    epochs=30,
+    validation_split=0.2,
+):
+    model.compile(
+        loss="sparse_categorical_crossentropy",
+        optimizer=optimizer,
+        metrics=["accuracy", "mae"],
+    )
+    history = model.fit(
+        X_train,
+        y_train,
+        batch_size=batch_size,
+        epochs=epochs,
+        validation_split=validation_split,
+    )
+
     # Evaluate the model on the test set
     test_loss, test_acc, test_mae = model.evaluate(X_test, y_test)
-    print(f'Test loss: {test_loss}, Test accuracy: {test_acc}, Test MAE: {test_mae}')
-    
+    print(f"Test loss: {test_loss}, Test accuracy: {test_acc}, Test MAE: {test_mae}")
+
     return history
 
+
 def evaluate_model(model, X_test, y_test):
     test_loss, test_acc, test_mae = model.evaluate(X_test, y_test)
-    print(f'Test loss: {test_loss}, Test accuracy: {test_acc}, Test MAE: {test_mae}')
-    
+    print(f"Test loss: {test_loss}, Test accuracy: {test_acc}, Test MAE: {test_mae}")
+
     return test_loss, test_acc, test_mae
 
-def reshape_data(X_train, X_test):
-    x_train_reshape = X_train.values.reshape(-1, 28, 28)
-    x_test_reshape = X_test.values.reshape(-1, 28, 28)
-    
-    return x_train_reshape, x_test_reshape
 
+def reshape_data(X_train, X_test):
+    x_train, x_test = X_train / 255.0, X_test / 255.0
+    x_train_flattened = x_train.reshape(x_train.shape[0], -1)
+    x_test_flattened = x_test.reshape(x_test.shape[0], -1)
+    return x_train_flattened, x_test_flattened
```

Comment:
Modified the reshape function to suit the data loaded from Keras. @awsactivators

Reply:
This is better, thanks